selectors/urls.js

```diff
@@ -0,0 +1,138 @@
+    getPatronFilter,
+    getServiceFilter,
+    getStatusFilter,
+    getHealthIssueFilter,
+    getNeedsFilter,
+    getSearchStringFilter,
+    getCurrentLanguage,
+    getSortBy
+  ],
+  (
+    patronFilter,
+    serviceFilter,
+    statusFilter,
+    healthIssueFilter,
+    selectedNeeds,
+    searchString,
+    currentLanguage,
+    sortBy
+  ) => {
+    let href = "/favourites";
+    href += "?lng=" + currentLanguage;
+    href += "&sortBy=" + sortBy;
+
+    if (Object.keys(selectedNeeds).length > 0) {
+      href += "&selectedNeeds=" + Object.keys(selectedNeeds).join();
+    }
+    if (patronFilter !== "") {
+      href += "&patronType=" + patronFilter;
+    }
+    if (serviceFilter !== "") {
+      href += "&serviceType=" + serviceFilter;
+    }
+    if (statusFilter !== "") {
+      href += "&statusAndVitals=" + statusFilter;
+    }
+    if (healthIssueFilter !== "") {
+      href += "&serviceHealthIssue=" + healthIssueFilter;
+    }
+    if (searchString !== "") {
+      href += "&searchString=" + searchString;
+    }
+    return href;
+  }
+);
+
+export const getPrintUrl = createSelector(
+  [
+    getFilteredBenefits,
+    getPatronFilter,
+    getServiceFilter,
+    getStatusFilter,
+    getHealthIssueFilter,
+    getNeedsFilter,
+    getSortBy,
+    getCurrentLanguage,
+    getClosestOffice,
+    getSelectedOffice,
+    getFromFavourites,
+    getFavoriteBenefits,
+    getBenefits
+  ],
+  (
+    filteredBenefits,
+    patronFilter,
+    serviceFilter,
+    statusFilter,
+    healthIssueFilter,
+    selectedNeeds,
+    sortBy,
+    language,
+    closestAreaOffice,
+    selectedAreaOffice,
+    fromFavourites,
+    favouriteBenefits,
+    benefits
+  ) => {
+    let filteredBenefitsIDs;
+    if (fromFavourites !== undefined) {
+      filteredBenefitsIDs = benefits
+        .filter(b => favouriteBenefits.indexOf(b.id) > -1)
+        .map(b => b.id);
+    } else {
+      filteredBenefitsIDs = filteredBenefits.map(b => b.id);
+    }
+    const needsIDs = Object.keys(selectedNeeds);
+
+    let url = "/print";
+    url += "?lng=" + language;
+    url += "&sortBy=" + sortBy;
+    if (filteredBenefitsIDs.length > 0) {
+      url += "&benefits=" + filteredBenefitsIDs.join(",");
+    }
+    if (patronFilter !== "") {
+      url += "&patronType=" + patronFilter;
+    }
+    if (serviceFilter !== "") {
+      url += "&serviceType=" + serviceFilter;
+    }
+    if (statusFilter !== "") {
+      url += "&statusAndVitals=" + statusFilter;
+    }
+    if (healthIssueFilter !== "") {
+      url += "&serviceHealthIssue=" + healthIssueFilter;
+    }
+    if (needsIDs.length > 0) {
+      url += "&needs=" + needsIDs.join(",");
+    }
+    if (closestAreaOffice.id !== undefined) {
+      url += "&closestAOID=" + closestAreaOffice.id;
+    }
+    if (selectedAreaOffice.id !== undefined) {
+      url += "&selectedAOID=" + selectedAreaOffice.id;
+    }
+    if (fromFavourites !== undefined) {
+      url += "&fromFavourites=" + fromFavourites;
+    }
```

Comment:
When modifying the url prop directly I believe we can do: `url.query['key'] = value`
or replace all `url.query = {key: value, ...}`. It would be nice if we could do something similar here to avoid some of the repetition.

Reply:
the problem is the passed arguments are not available as an iterable object, I looked into that before creating the above mess

Reply:
I guess you can https://www.electrictoolbox.com/loop-parameters-passed-javascript-function/

Reply:
I was thinking more along the lines of building a `query` object with everything we want in the url, and then making another function that takes `query` as an argument and returns the url.

Reply:
That probably make sense - now that it is in a single place, it should be a quick refactor, I will create an issue